src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Directory.Move.cs

```diff
@@ -16,11 +20,41 @@ public void Directory_Move_To_Same_Directory()
         }
 
         [Fact]
+        [PlatformSpecific(TestPlatforms.OSX)]
         public void Directory_Move_From_Watched_To_Unwatched()
         {
             DirectoryMove_FromWatchedToUnwatched(WatcherChangeTypes.Deleted);
         }
 
+        [Theory]
+        [PlatformSpecific(TestPlatforms.OSX)]
+        [InlineData(1)]
+        [InlineData(2)]
+        [InlineData(3)]
+        public void Directory_Move_Multiple_From_Watched_To_Unwatched_Mac(int filesCount)
+        {
+            DirectoryMove_Multiple_FromWatchedToUnwatched(filesCount, true);
```

Comment:
Nit: please name Boolean args, e.g. `skipOldEvents: true`

Reply:
Since this is platform dependent, can we handle skipOldEvents inside the helper instead of passing it in?

Reply:
The helper is quite complex and besides this filter it is the same for all platforms. I actually don't like the concept of those filters as the test is just calling a method. Instead the test should do the test itself.

My proposal there is:

```csharp
testcase()
{
    // the collectEvents method is platform independent and don't care about test at all. It perform file changes and transform events to data.
    var collectedEvents = CollectEvents_Move_Multiple_FromWatchedToUnwatched(filesCount);
    // this is tricky as we don't know file names used in CollectEvents method. Maybe CollectEvents can return those names.
    var expectedEvents = ...

    // Finally the test do the test
    Assert.Equals(collectedEvents, expectedEvents)
}
```

or with more separation which I actually like more it would be:

```csharp
testcase()
{
    var filesToMove = getFileOperations(filesCount)

    var collectedEvents = CollectEvents_Move_Multiple_FromWatchedToUnwatched(filesToMove);
    
    var expectedEvents = exctract from filesToMove

    // Finally the test do the test
    Assert.Equals(collectedEvents, expectedEvents)
}
```

However, both are far away from what is in the test now. I would consider a new ticket for such a change which can cover and simplified current tests.